app/tests/test_user.py

```diff
@@ -0,0 +1,49 @@
+from ..models import User
+from ..serializers import UserSerializer
+from .base import BaseTestCase
+
+
+class UserTestCase(BaseTestCase):
+
+    def test_get_all_users(self):
+        response = self.client.get('/users/')
+        self.assertEqual(response.status_code, 200)
+
+    def test_retrieve_valid_single_user(self):
+        response = self.client.get('/users/{}/'.format(self.user.pk))
+        user = User.objects.get(pk=self.user.user_id)
+        serializer = UserSerializer(user)
+        self.assertEqual(response.data, serializer.data)
+        self.assertEqual(response.status_code, 200)
+
+    def test_retrieve_invalid_single_user(self):
+        response = self.client.get('/users/10/')
+        self.assertEqual(response.status_code, 404)
+
+    def test_create_new_user(self):
+        response = self.client.post('/users/', {
+            'role': [self.role.id],
+            'display_name': 'test user',
+            'about': 'test bio',
+            'avatar': 'testimage.png',
+            'pronoun': self.pronoun.id,
+            'skills': [self.skill.id],
+            'spoken_languages': [self.spoken_language.id],
+            'timezone': 'Africa/Lagos'
+        })
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(User.objects.count(), 2)
+
+    def test_update_fields_in_single_user(self):
```

Comment:
I think you should also test updating list fields

Reply:
lol sigh, okay.